junit-jupiter-api/src/main/java/org/junit/jupiter/api/TestFactory.java

```diff
@@ -34,6 +34,9 @@
  * will then be executed lazily, enabling dynamic and even non-deterministic
  * generation of test cases.
  *
+ * <p>Any Stream returned by a {@code @TestFactory} will be properly closed by calling {@code stream.close()},
```

Comment:
Minor nitpick - please consider surrounding `Stream` with `{@code ... }` for consistency with all the other class references. :)